Source/WebKit/UIProcess/Extensions/API/WebExtensionContextAPIDeclarativeNetRequest.cpp

```diff
@@ -171,7 +171,11 @@ void WebExtensionContext::declarativeNetRequestUpdateDynamicRules(String&& rules
         return ruleID;
     });
 
-    auto rulesToAdd = JSON::Value::parseJSON(rulesToAddJSON)->asArray();
+    Ref<JSON::Array> rulesToAdd = JSON::Array::create();
+    if (!rulesToAddJSON.isEmpty()) {
+        if (RefPtr<JSON::Value> parsedJSON = JSON::Value::parseJSON(rulesToAddJSON); parsedJSON->asArray())
+            rulesToAdd = *(parsedJSON->asArray());
```

Comment:
Might as well avoid calling `asArray()` twice.

```suggestion
        if (RefPtr parsedJSON = JSON::Value::parseJSON(rulesToAddJSON)) {
            if (RefPtr array = parsedJSON->asArray())
                rulesToAdd = *array;
        }
```